### DISCUSSION TITLE
ESP32/ESP-32S Development Board NodeMCU-32S CH340 MICRO USB compatibility

### DISCUSSION THREAD

**Discussion by maxi1134** [ESPresense/ESPresense#1373] [General]
I am about to order over 10 https://www.aliexpress.com/item/1005006068563165.html?spm=a2g0o.productlist.main.63.37292f2eT8WfQB (ESP32/ESP-32S Development Board NodeMCU-32S CH340 MICRO USB WiFi+For Bluetooth UltraLow Power Consumption Dual Core  ESP32-WROOM) and would like to know if I can expect them to work with ESPRESENSE, I currently use regular esp32 dev boards, but their wifi retry rates is at 40-50%.

I wish to switch to the NODEMCU as they have an external antenna.

**Comment by dxmnkd316** [comment]
I don't think you're going to get a guarantee they'll work.  Maybe.  But unless they're on the official list, it's kind of hit or miss.  I spent an extra couple bucks on shipping to prevent having 10+ boards that didn't work for the application I wanted.

When I first started down the route of ESPresense, I started with one board, verified it worked, and then ordered the rest.  I wasn't going to spend $100 on boards and power supplies I wasn't sure would work or didn't give me the performance I wanted.

I used [this board](https://www.digikey.com/en/products/detail/espressif-systems/ESP32-DEVKITC-32E/12091810) successfully, but I eventually swapped nearly all of them out for [m5stack Atom Lite S3.](https://www.digikey.com/en/products/detail/m5stack-technology-co-ltd/C124/18070571)  The accuracy seemed to be much better with the m5Stack board (but this is probably just selection bias on my part).  Anyways, the ESP32-DEVKITC-32E (ESP32-WROOM-32E 4MB FLASH) worked well.  I still use it for enrolling Apple devices.  The S3 doesn't seem to enroll very well but it did have the advantage of a cleaner case, being cheaper, and it has a Grove port.

**Comment by maxi1134** [comment]
I confirm [these ](https://www.amazon.ca/dp/B0CH9JKBC3 )work!

I ordered 4 and no issue.
